src/understand-tidb/session.md

```diff
@@ -1 +1,122 @@
 # Session
+
+The `session` package (and related packages such as `sessionctx` and `variable`) are responsible for maintaining the state of both sessions and transactions.
+
+## New session origins
+
+New connections are first established in the `server` package. After some initial protocol negotiation, the `server` package [calls](https://github.com/pingcap/tidb/blob/246c59d5c926780235dd25aef27c469ffe376e21/server/driver_tidb.go#L181-L186) `session.CreateSession()`. This function then calls `session.createSessionWithOpt()` (via `CreateSessionWithOpt()`) which creates the session.
+
+Sessions used for internal server operations are [usually created in a different manner](https://github.com/pingcap/tidb/blob/649ed6abc9790cfdd2a17065118379d8abcc7595/executor/simple.go#L83-L93), with the sessionctx being retrieved from a pool of sessions maintained by `domain`:
+```go
+dom := domain.GetDomain(e.ctx)
+sysSessionPool := dom.SysSessionPool()
+ctx, err := sysSessionPool.Get()
+if err != nil {
+    return nil, err
+}
+restrictedCtx := ctx.(sessionctx.Context)
+restrictedCtx.GetSessionVars().InRestrictedSQL = true
+```
+
+Internal sessions will not show up in the output of `SHOW PROCESSLIST`, and because they do not have a privilege manager handle attached skip all privilege checks.
```

Comment:
I've written this a bit differently:
```
Internal sessions will not show up in the output of `SHOW PROCESSLIST`, and skip all privilege checks because they do not have a privilege manager handle attached.
```